docs/architecture/language-server.md

```diff
@@ -0,0 +1,133 @@
+# Language Server
+
+This document outline the architecture of the Puya Language Server capability.
+
+```mermaid
+sequenceDiagram
+  actor D as Developer
+  participant IDE as IDE
+  participant LC as Language Client
+  participant LS as Language Server
+  participant PF as Puya Frontend
+  participant PB as Puya Backend
+  D ->> IDE: keystroke
+  D ->> IDE: vcsoperations
+  IDE ->> LC: documentOpened (triggers init when not started)
+  IDE ->> LC: configurationChanged (triggers reinit)
+  IDE ->> LC: restartLanguageServer (triggers reinit)
+  LC ->>+ LS: initialise()
+  LS -->>- LC: initialised
+  IDE ->> LC: documentChanged
+  LC ->>+ LS: documentChanged
+  LS ->>+ PF: compile(project)
+  PF -->> LS: diagnostics
+  PF -->>- LS: awst
+  LS ->>+ PB: compile(awst)
+  PB -->>- LS: diagnostics
+  LS -->>- LC: diagnostics
+```
+
+## Algorand Python Language Server
+
+The Python implementation of the Puya language server directly interacts with the Puya compiler components in the same Python process.
+
+```mermaid
+flowchart TB
+    subgraph "VSCode IDE"
+        Developer["Developer"]
+        VSCodeExt["VSCode Extension"]
+    end
+    
+    subgraph "Python Process"
+        PuyaPyLSP["PuyaPy Language Server (puyapy.lsp)"]
+        
+        subgraph "PuyaPy Frontend"
```

Comment:
If i'm not mistaken, both the PuayPy and PuyaTs frontends can also return diagnostics when it fails to produce a valid AWST which will be surfaced by the language server?